docs/SIL.rst

```diff
@@ -2166,6 +2166,48 @@ requires ``release`` or other cleanup, that value must be loaded before being
 stored over and cleaned up. It is undefined behavior to store to an address
 that points to deallocated storage.
 
+load_borrow
+```````````
+
+::
+
+   sil-instruction ::= 'load_borrow' sil-value
+
+   %1 = load_borrow %0 : $*T
+   // $T must be a loadable type
+
+Loads the value ``%1`` from the memory location ``%0``. The ``load_borrow``
+instruction creates a borrowed scope in which a read-only borrow value ``%1``
```

Comment:
You should define what's a read-only borrow value is. I assume it's the same as for end_borrow (except %1 instead of %0):

1. If ``%1`` is an address, ``%1`` can not be written to.
2. If ``%1`` is a non-trivial value, ``%1`` can not be destroyed.